crypto/dh/dh_check.c

```diff
@@ -57,13 +57,8 @@ int DH_check_params(const DH *dh, int *ret)
     nid = DH_get_nid((DH *)dh);
     if (nid != NID_undef)
         return 1;
-    /*
-     * OR
-     * (2b) FFC domain params conform to FIPS-186-4 explicit domain param
-     * validity tests.
-     */
-    return ossl_ffc_params_FIPS186_4_validate(dh->libctx, &dh->params,
-                                              FFC_PARAM_TYPE_DH, ret, NULL);
+
+    return 0;
```

Comment:
Is there a reason to remove this check in non-FIPS mode?

Reply:
This code does not exist in non-fips code path. Ifdef starts on line 47.

Is this a request to add this check to default provider for the first time?

Reply:
I didn't see there was an other DH_check_params() in case of non-FIPS. Adding such a new check looks like it's out of scope of this PR, and it's a deprecated function without internal callers anyway.